### PR TITLE
chore: add a workflow that tracks acvm version for integration tests

### PR DESCRIPTION
**Pull request by jonybur**
# Description

Adds a workflow that checks that the version of the acvm of the integration test matches the acvm version specified in the Cargo.lock

## Problem\*

Resolves https://github.com/noir-lang/noir/issues/2706

## Documentation

- [ ] This PR requires documentation updates when merged.

  

  - [ ] I will submit a noir-lang/docs PR.

  

  - [ ] I will request for and support Dev Rel's help in documenting this PR.

  
  

## Additional Context

# PR Checklist\*

- [ ] I have tested the changes locally.
- [ ] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

**Comment by TomAFrench** [comment]
Wouldn't it be easier to just have a check that `cargo.lock` and `yarn.lock` point to the same version and leave it to the developer to update the `acvm_js` dependency?

File: .github/workflows/track-acvm.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+          CURRENT_VERSION=$(awk '/name = "acvm"/ {getline; print $3}' Cargo.lock | tr -d '"')
+
+          # Extract acvm version from the previous commit's Cargo.lock
+          PREVIOUS_VERSION=$(git show HEAD~1:Cargo.lock | awk '/name = "acvm"/ {getline; print $3}' | tr -d '"')
+
+          echo "Current ACVM Version: $CURRENT_VERSION"
+          echo "Previous ACVM Version: $PREVIOUS_VERSION"
+
+          if [ "$CURRENT_VERSION" != "$PREVIOUS_VERSION" ]; then
+            jq '.dependencies["@noir-lang/acvm_js"] = "^'${CURRENT_VERSION}'"' compiler/integration-tests/package.json > tmp.json && mv tmp.json compiler/integration-tests/package.json
```
Comment (TomAFrench) [inline .github/workflows/track-acvm.yml:31]: `yarn.lock` would need to be refreshed if we're going to be making this commit.

File: .github/workflows/track-acvm.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Check acvm version change
+        run: |
+          # Extract acvm version from the current Cargo.lock
+          CURRENT_VERSION=$(awk '/name = "acvm"/ {getline; print $3}' Cargo.lock | tr -d '"')
+          INTEGRATION_TEST_VERSION=$(jq '.dependencies["@noir-lang/acvm_js"]' compiler/integration-tests/package.json | tr -d '"' | sed 's/^[^0-9]*//')
```
Comment (TomAFrench) [inline .github/workflows/track-acvm.yml:19]:
```suggestion
          INTEGRATION_TEST_VERSION=$(yarn workspace integration-tests info @noir-lang/acvm_js --json | jq .children.Version | tr -d '"')
```

This should be equivalent but doesn't require me to think too hard about regexes.

File: .github/workflows/track-acvm.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
+name: Track ACVM for Integration Tests
+
+on:
+  push:
+    branches:
+      - '*'
```
Comment (TomAFrench) [inline .github/workflows/track-acvm.yml:6]: Would be nice to restrict this to when `Cargo.lock` or `yarn.lock` changes. Not sure if we can do an OR on path triggers though (should be able to).

**Review by TomAFrench — COMMENTED** [review]
Looks good, couple of nits

File: .github/workflows/track-acvm.yml (hunk 3 above)
Comment (jonybur) [reply .github/workflows/track-acvm.yml:6]: Revised

**Review by TomAFrench — DISMISSED** [review]
LGTM

**Review by kobyhallx — APPROVED** [review]

File: compiler/integration-tests/package.json [hunk 4]
```diff
@@ -12,7 +12,7 @@
   },
   "dependencies": {
     "@aztec/bb.js": "^0.5.1",
```
Comment (kevaundray) [inline compiler/integration-tests/package.json:14]:
```suggestion
    "@aztec/bb.js": "^0.6.7",
```

**Review by TomAFrench — APPROVED** [review]

**Pull request merged**